src/server/utils.spec.js

```diff
@@ -0,0 +1,56 @@
+import {
+  getQueryKeys,
+  hasMeasurePrefix,
+  createGroupAggregation,
+} from './utils';
+
+const a = 'count';
+const b = 'month';
+const c = 'measure-potato';
+const d = 'potato';
+const keys = [a, b, c, d];
+
+const data = {
+  operation: {
+    selectionSet: {
+      selections: [
+        {
+          selectionSet: {
+            selections: [
+              { name: { value: a } },
+              { name: { value: b } },
+              { name: { value: c } },
+              { name: { value: d } },
+            ],
+          },
+        },
+      ],
+    },
+  },
+};
+
+describe('#getQueryKeys', () => {
+  it('should get the keys', () => {
+    expect(getQueryKeys(data)).toEqual(keys);
+  });
+
+  // This was one thing the functional implementation helped with
```

Comment:
`s/helped/hurt`. I'm not a fan of swallowing errors -- I'm of the opinion that if I pass in a malformed object, it _should_ blow up.